File: fathersDay/Dsa/Day 34 (1791. Find Center of Star Graph)/Solution.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

class Solution {
public:
    int CenterOfStarGraph(std::vector<std::vector<int>>& edges) {
        std::unordered_map<int, int> degree;
        for (const auto& edge : edges) {
            degree[edge[0]]++;
            degree[edge[1]]++;
        }
        int n = edges.size() + 1;
        for (const auto& pair : degree) {
            if (pair.second == n - 1) {
                return pair.first;
            }
        }
        return -1;
    }
};

int main() {
    Solution solution;
    std::vector<std::vector<int>> edges = {{1, 2}, {2, 3}, {4, 2}};
    int result = solution.CenterOfStarGraph(edges);
    std::cout << result << std::endl; // Output: 2
    return 0;
}

File: fathersDay/Dsa/Day 34 (1791. Find Center of Star Graph)/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int CenterOfStarGraph(int[][] edges) {
        Map<Integer, Integer> degree = new HashMap<>();
        for (int[] edge : edges) {
            degree.put(edge[0], degree.getOrDefault(edge[0], 0) + 1);
            degree.put(edge[1], degree.getOrDefault(edge[1], 0) + 1);
        }
        int n = edges.length + 1;
        for (Map.Entry<Integer, Integer> entry : degree.entrySet()) {
            if (entry.getValue() == n - 1) {
                return entry.getKey();
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[][] edges = {{1, 2}, {2, 3}, {4, 2}};
        int result = solution.CenterOfStarGraph(edges);
        System.out.println(result); // Output: 2
    }
}
